### DISCUSSION TITLE
How do we get all the child cells of a subcell

### DISCUSSION THREAD

**Discussion by joamatab** [gdsfactory/kfactory#509] [Q&A]
c.each_child_cell() does not return all the cells in the hierarchy

how do we get all the cell names in the hiearchy?

```python
        for cell_index in c.each_child_cell():
            ci = c.kcl[cell_index]
            print(ci.name)
```

**Comment by joamatab** [comment]
is `[ci.name for ci in c.kcl.kcells.values()]`   the prefered way?

@sebastian-goeldi

**Comment by sebastian-goeldi** [comment]
what are all the child cells in the hierarchy?  you mean all cells which are in any hierarchy level below the cell? If yes, then https://klayout.de/doc/code/class_Cell.html#method22 i.e. `[c.kcl[cc] for cc in c.called_cells()]`

**Reply by joamatab** [reply]
all the child cells for a particular device

the issue with `[ci.name for ci in c.kcl.kcells.values()]` is that it can be slow

**Reply by sebastian-goeldi** [reply]
Of course iterating through all cells is slow. Therefore the one above
